torch_geometric/nn/dense/linear.py

```diff
@@ -258,64 +259,110 @@ def reset_parameters(self):
         reset_bias_(self.bias, self.in_channels,
                     self.kwargs.get('bias_initializer', None))
 
+    @torch.jit.unused
+    def forward_segmm(self, x: Tensor, type_vec_ptr: Tensor) -> Tensor:
+        assert self.weight is not None
+        out = pyg_lib.ops.segment_matmul(x, type_vec_ptr, self.weight)
+        return out
+
+    def forward_naive(self, x: Tensor, type_vec_ptr: Tensor) -> Tensor:
+        out = x.new_empty(x.size(0), self.out_channels)
+
+        for i in range(self.num_types):
+            off_start, off_end = type_vec_ptr[i], type_vec_ptr[i + 1]
+            subset_out = x[off_start:off_end] @ self.weight[i]
+            # The data type may have changed with mixed precision:
+            out[off_start:off_end] = subset_out.to(out.dtype)
+
+        return out
+
+    @torch.jit.unused
+    def _update_timing_cache(self, x: Tensor, type_vec_ptr: Tensor,
+                             num_rows: int) -> bool:
+        measure_iter = 3
```

Comment:
nit: I'm in favor of putting this as a constant outside the class so that one can easily experiment this.

Reply:
agreed